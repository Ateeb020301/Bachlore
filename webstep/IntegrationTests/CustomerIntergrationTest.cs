using System;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;
using Google;
using webstep.Data;
using webstep.GraphQL.Entities;
using webstep.GraphQL.Mutations;
using webstep.GraphQL;
using webstep.GraphQL.NodaTime.Types;
using webstep.Models;

namespace GraphQL.Tests
{
    public class CustomerIntergrationTest
    {
        [Fact]
        [Obsolete]
        public async Task Customer_Schema_Changed()
        {

            // arrange
            // act
            ISchema schema = await new ServiceCollection()
                .AddPooledDbContextFactory<WebstepContext>(
                    options => options.UseInMemoryDatabase($"Data Source=conferences-{Guid.NewGuid()}.db"))
                .AddGraphQL()
                .AddSorting()
                .AddFiltering()
                .AddProjections()  // Add Projections
                .AddType<LocalDateType>()
                .AddQueryType<Query>()
                .AddMutationType(d => d.Name("Mutation"))
                    .AddTypeExtension<CustomerMutation>()
                .AddType<CustomerType>()
                .EnableRelaySupport()
                .BuildSchemaAsync();

            // assert
            schema.Print().MatchSnapshot();
        }
        [Fact]
        [Obsolete]
        public async Task RegisterCustomer()
        {
            // arrange
            IRequestExecutor executor = await new ServiceCollection()
                .AddPooledDbContextFactory<WebstepContext>(
                    options => options.UseInMemoryDatabase($"Data Source=conferences-{Guid.NewGuid()}.db"))
                .AddGraphQL()
                .AddSorting()
                .AddFiltering()
                .AddProjections()  // Add Projections
                .AddType<LocalDateType>()
                .AddQueryType<Query>()
                .AddMutationType(d => d.Name("Mutation"))
                    .AddTypeExtension<CustomerMutation>()
                .AddType<CustomerType>()
                .EnableRelaySupport()
                .BuildRequestExecutorAsync();

            // act
            IExecutionResult result = await executor.ExecuteAsync(@"
            mutation {
                addCustomer(input: {
                    firstName: ""John"",
                    lastName: ""Doe"",
                    email: ""johndoe@example.com"",
                    adresse: ""123 St"",
                    tlf: ""1234567890""
                }) {
                customer {
                    firstName
                    lastName
                    email
                    adresse
                    tlf
                }
                }
            }");

            // assert
            result.ToJson().MatchSnapshot();
        }
    }
}