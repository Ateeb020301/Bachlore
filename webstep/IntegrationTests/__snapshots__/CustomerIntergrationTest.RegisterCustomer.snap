{
  "errors": [
    {
      "message": "\u0060sellerId\u0060 is a required field and cannot be null.",
      "locations": [
        {
          "line": 3,
          "column": 36
        }
      ],
      "path": [
        "addCustomer"
      ],
      "extensions": {
        "field": "sellerId",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Input-Object-Required-Fields"
      }
    }
  ]
}
